edx_analytics_transformers/transformers/base_transformer.py

```diff
@@ -0,0 +1,61 @@
+"""
+Base transformer to add or transform common data values.
+"""
+import json
+from logging import getLogger
+
+import six
+
+
+logger = getLogger(__name__)
+
+
+class BaseTransformer:
+    """
+    Base transformer class.
+
+    Other transformers are inherited from this class.
+    """
+    required_fields = ()
+    additional_fields = ()
```

Comment:
`required_fields` is added in the standard's base transformer (e.g. caliper transformer. in that case `action`, `object`, `type` are the required fields for every event). But for the keys that are specific to some transformers/events (e.g. `target` is specific to some video events) we can use the `additional_fields` in the child transformer. 

Otherwise we would have to do something like this in child transformer:
```
required_fields = CaliperTransformer.required_fields + ('target',)
```

This way, we just have to specify the optional fields in event transformers.